### ISSUE TITLE
Disable glider when taken damage

### ISSUE THREAD

**Issue by lincore81** [Tictim/Paraglider#19]
In my modpack I added lava as a wind source, which is a really fun way of exploring the nether. I was hoping ghasts, bone serpents from Alex' Mobs etc. would make this a somewhat dangerous manoeuvre, but I can just tank their damage and glide to safety.

TL;DR I suggest the glider gets disabled when the player takes damage mid-glide. Players should not be able to redeploy it until they hit the ground or fell quite a distance. I think this could also spice up PVP a bit (although shooting down paratroopers is against the Geneva Convention :). This probably should not be the default behaviour, but a config option would be great.

**Comment by Tictim** [comment]
I agree with you that the Paraglider makes the nether somewhat less hazardous, even without lava as wind source. Disabling paragliding when taking damage sounds simple and straightforward too.

Making all damage to interrupt paragliding wouldn't cut it, though. What's fun with damage from starvation knocking off of you from paragliding? Also, from my experience, jumping into fire/campfire with Paraglider makes one or two hits from flame before taking you upward, so that should be taken into account... Or not, because this kind of tiny damages are everywhere and there's absolutely no way to blacklist such damage types manually. Not to mention this approach doesn't work well with other mods.

One alternative approach would be using knockback instead of damage. Not all damages generate knockback, and not all knockbacks come with damage. But it sounds reasonable to assume the "knockback" part should effect paragliding, not the damage itself.

There is an event called `LivingKnockBackEvent` fired whenever knockback is applied to someone, so at least implementing it is easy. But if there's any "knockbacks" or push-offs that doesn't use this event, it will create an oddity. I'm assuming there's a LOT of it, and half of them doesn't even have a way to hook into those things. We simply have to test it out, though.

Another approach would be setting minimum amount of damage required to ignore trivial damages. It's very sketchy indeed. and might not be something we desired. But I mean, it's an option.

Duration of interruption should be considered. Getting plunged to death because got hit once sounds quite unfair, so I think players should be able to recover after some time. But how much? Super long recover time is just equal to no recover at all. Short recover time sounds not enough to affect anything. But players will experience massive rate of height loss without paragliding. Any interruption can make players diverge from trajectory they expected.

I think recover time should be kept to minimum to not induce any unfair deaths, while still affecting gameplay enough to make players to consider getting hit as nontrivial threat. But it's just my opinion after all. Sure, I could make config option to give players the power of increasing/decreasing recover time for their needs, but deciding what default value should be is still valuable. Also I hate the idea of "Balance it yourself" mod.

tl;dr cool idea but needs more design input.

**Comment by NewbulaCo** [comment]
Dropping by because I had the same suggestion; I think it would be a great config to have. I think having any damage regardless of type would be the way to do make the damage cooldown function. Taking damage from starvation already means you're not able to run, so lacking the strength to paraglide seems in line with that. That also falls in line with being poisoned. To avoid taking fire damage, timing the jump or making sure you're a bit higher would guarantee you can take off... you can also use campfires. A damage cooldown should be long enough to give you a bit of fall damage, but not enough to outright kill you, so 5-6 blocks with the glider would make sense. Also considering that this optional config is to encourage a more challenging experience, this type of setup makes sense to me... but that's just my opinion lmao, thanks for reading
